src/main.rs

```diff
@@ -122,23 +115,77 @@ fn main() {
             debug!("Path updated: {:?}", path);
         }
 
-        // Wait for current child process to exit
-        wait_process(&child_process, kill, args.restart);
+        // We have three scenarios here:
+        //
+        // 1. Make sure the previous run was ended, then run the command again
+        // 2. Just send a specified signal to the child, do nothing more
+        // 3. Send SIGTERM to the child, wait for it to exit, then run the command again
+        // 4. Send a specified signal to the child, wait for it to exit, then run the command again
+        //
+        match args.restart {
```

Comment:
Let's make a tuple here based on `args.restart` and `signal` so we can match on exactly the cases we want and let the compiler ensure we're handling everything.